samcli/local/apigw/local_apigw_service.py

```diff
@@ -18,35 +18,85 @@
 
 
 class Route(object):
-
-    def __init__(self, methods, function_name, path, binary_types=None, stage_name=None, stage_variables=None):
+    ANY_HTTP_METHODS = ["GET",
+                        "DELETE",
+                        "PUT",
+                        "POST",
+                        "HEAD",
+                        "OPTIONS",
+                        "PATCH"]
+
+    def __init__(self, function_name, path, method):
         """
         Creates an ApiGatewayRoute
 
         :param list(str) methods: List of HTTP Methods
         :param function_name: Name of the Lambda function this API is connected to
         :param str path: Path off the base url
         """
-        self.methods = methods
+        self.method = method.upper()
         self.function_name = function_name
         self.path = path
-        self.binary_types = binary_types or []
-        self.stage_name = stage_name
-        self.stage_variables = stage_variables
+
+    @staticmethod
+    def normalize_routes(routes):
+        """
+        Normalize the Routes to use standard method name
+        Parameters
+        ----------
+        route : list of samcli.commands.local.apigw.local_apigw_service.Route
+            List of Routes to replace normalize
+        Returns
+        -------
+        list of list of samcli.commands.local.apigw.local_apigw_service.Route
+            List of normalized Routes
+        """
+
+        result = list()
+        for route in routes:
+            for normalized_method in Route.normalize_http_methods(route.method):
+                result.append(Route(method=normalized_method, function_name=route.function_name, path=route.path))
+        return result
+
+    def __eq__(self, other):
+        return isinstance(other, Route) and \
+               self.method == other.method and self.function_name == other.function_name and self.path == other.path
+
+    def __hash__(self):
+        return hash(self.function_name) * hash(self.path) * hash(self.method)
+
+    @staticmethod
+    def normalize_http_methods(http_method):
```

Comment:
`ANY` is an API Gateway specific 'verb'. I don't think this belongs in the Route class. The Route is meant to be generic and follow valid HTTP routes/methods.